File: easy/find_index_first_occurence.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for i in range(len(haystack)-len(needle)+1):
            # print(haystack[i:i+len(needle)])
            if haystack[i:i+len(needle)] == needle:
                return i
        return -1

File: easy/find_index_first_occurence.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  for (let i = 0; i < haystack.length - needle.length + 1; i++) {
    // console.log(i)
    // console.log(haystack.substr(i,needle.length))
    if (haystack.substr(i, needle.length) === needle) {
      return i;
    }
  }
  return -1;
};
